File: Collector/OLD/LegacyJava/Collector/Dimension/World.java
package Collector.Dimension;

import Collector.Character.Player;

import static Collector.Restrictions.*;

public class World {
    public static void generateWorld(int x, int y) {
        if (Chunks.isEmpty(x, y)) {
            Chunks.generateChunk(x, y);
        }
    }

    public static void ungenerateWorld(int x, int y) {
        if (!Chunks.isEmpty(x, y)) {
            Chunks.ungenerateChunk(x, y);
        }
    }

    public static void loadChunks() {
        for (int i = -(RENDER_DISTANCE); i < RENDER_DISTANCE; i++) {
            for (int j = -(RENDER_DISTANCE); j < RENDER_DISTANCE; j++) {
                generateWorld(
                        i + Player.getX() / (TILE_SIZE * SUPER_CHUNK_SIZE * CHUNK_SIZE),
                        j + Player.getY() / (TILE_SIZE * SUPER_CHUNK_SIZE * CHUNK_SIZE)
                );
            }
        }
    }

    public static void unloadChunks() {
        for (int i = -RENDER_DISTANCE; i < RENDER_DISTANCE+1; i++) {
            //Down
            ungenerateWorld(
                    Player.getX() / (TILE_SIZE * SUPER_CHUNK_SIZE * CHUNK_SIZE)+i,
                    (Player.getY() / (TILE_SIZE * SUPER_CHUNK_SIZE * CHUNK_SIZE))+3
            );
            //Up
            ungenerateWorld(
                    Player.getX() / (TILE_SIZE * SUPER_CHUNK_SIZE * CHUNK_SIZE)+i-1,
                    (Player.getY() / (TILE_SIZE * SUPER_CHUNK_SIZE * CHUNK_SIZE))-4
            );
            //Right
            ungenerateWorld(
                    Player.getX() / (TILE_SIZE * SUPER_CHUNK_SIZE * CHUNK_SIZE)-4,
                    (Player.getY() / (TILE_SIZE * SUPER_CHUNK_SIZE * CHUNK_SIZE))+i
            );
            //Left
            ungenerateWorld(
                    Player.getX() / (TILE_SIZE * SUPER_CHUNK_SIZE * CHUNK_SIZE)+3,
                    (Player.getY() / (TILE_SIZE * SUPER_CHUNK_SIZE * CHUNK_SIZE))+i
            );
        }

    }
}

File: Collector/src/Dimension/World.cs
using System;
using Collector.Character;

namespace Collector.Dimension
{
    public class World {
        private void GenerateWorld(float x, float y)
        {
            if (Chunks.IsEmpty(x, y)) {
                Chunks.GenerateChunk(x, y);
            }
        }

        private void UngenerateWorld(float x, float y) {
            if (!Chunks.IsEmpty(x, y)) {
                Chunks.UngenerateChunk(x, y);
            }
        }

        public void LoadChunks()
        {
            for (var i = -(IRestrictions.RenderDistance); i < IRestrictions.RenderDistance; i++) {
                for (var j = -(IRestrictions.RenderDistance); j < IRestrictions.RenderDistance; j++) {
                    GenerateWorld(
                        i + Player.X / (IRestrictions.SuperChunkSize * IRestrictions.ChunkSize),
                        j + Player.Y / (IRestrictions.SuperChunkSize * IRestrictions.ChunkSize)
                    );
                }
            }
        }

        public void UnloadChunks() {
            for (var i = -IRestrictions.RenderDistance; i < IRestrictions.RenderDistance+1; i++) {
                //Down
                UngenerateWorld(
                    Player.X / (IRestrictions.SuperChunkSize * IRestrictions.ChunkSize)+i,
                    Player.Y / (IRestrictions.SuperChunkSize * IRestrictions.ChunkSize)+IRestrictions.RenderDistance
                );
                //Up
                UngenerateWorld(
                    Player.X / (IRestrictions.SuperChunkSize * IRestrictions.ChunkSize)+i,
                    Player.Y / (IRestrictions.SuperChunkSize * IRestrictions.ChunkSize)-IRestrictions.RenderDistance
                );
                //Right
                UngenerateWorld(
                    Player.X / (IRestrictions.SuperChunkSize * IRestrictions.ChunkSize)-IRestrictions.RenderDistance,
                    Player.Y / (IRestrictions.SuperChunkSize * IRestrictions.ChunkSize)+i
                );
                //Left
                UngenerateWorld(
                    Player.X / (IRestrictions.SuperChunkSize * IRestrictions.ChunkSize)+IRestrictions.RenderDistance,
                    Player.Y / (IRestrictions.SuperChunkSize * IRestrictions.ChunkSize)+i
                );
            }
        }
    }
}
